### PR TITLE
adds maven for dep search

### PR DESCRIPTION
**Pull request by agzam**
Please answer the following questions and leave the below in as part of your PR.

- [ ] This PR corresponds to an #41 

- [ ] I have updated the [CHANGELOG.md](https://github.com/babashka/neil/blob/main/CHANGELOG.md) file with a description of the addressed issue.

*Aww. Only after figuring this out I just realized there's #162 with a comment. Just following and grabbing that code would've been maybe simpler* 🤦

## Tested with the following:

```
$> ./neil dep search clojure
./neil dep search clojure
;; Tons of results from both Maven and Clojars

$> ./neil dep search "org.clojure/clojurescript"

./neil dep search "org.clojure/clojurescript"
Unable to find org.clojure/clojurescript on Clojars.
:lib org.clojure/clojurescript :version "1.11.121" :description "org.clojure/clojurescript on Maven"


$> ./neil dep search "org.clojure/clojure"

./neil dep search "org.clojure/clojure"
Unable to find org.clojure/clojure on Clojars.
:lib org.clojure/clojure-install :version "0.1.21" :description "org.clojure/clojure-install on Maven"
:lib org.clojure/clojure-contrib :version "1.2.0" :description "org.clojure/clojure-contrib on Maven"
:lib org.clojure/clojure :version "1.12.0-alpha5" :description "org.clojure/clojure on Maven"


$> ./neil dep search "clojurefoo"
./neil dep search "clojurefoo"
Unable to find clojurefoo on Maven.
Unable to find clojurefoo on Clojars.


$> ./neil dep search ""
;; produces help info

$> ./neil dep search 123
;; produces help info

$> ./neil dep search '123'
;; produces help info
```

## Also neil.el was adjusted for the changes

### CODE REVIEW

File: src/babashka/neil.clj [hunk 1]
```diff
@@ -500,17 +500,46 @@ will return libraries with 'test framework' in their description.
...
+             "https://search.maven.org/solrsearch/select?q=a:%s&rows=20&wt=json"
+             (url-encode search-term))
+        keys-m {:g :group_name
+                :a :jar_name
+                :timestamp :created
+                :latestVersion :version}
+        res (-> url curl-get-json :response :docs
+                first
+                (clojure.set/rename-keys keys-m)
+                (select-keys (vals keys-m)))]
```
Comment (borkdude) [inline src/babashka/neil.clj:514]: select-keys always returns a map so `res` is never nil

**Comment by borkdude** [comment]
If this is still a work in progress, can you mark the PR as draft? And then ping me for a review when done?

**Comment by agzam** [comment]
> If this is still a work in progress, can you mark the PR as draft? And then ping me for a review when done?

Apologies for constantly buzzing you with change notifications. I thought I was done, but then I realized that Emacs counterpart needs updating too.
Seemingly small change turned out to be a bit tricky. It's all good now - I checked it with different inputs. Thank you for your patience.

**Comment by borkdude** [comment]
Great, I'll have a look tomorrow!

**Comment by borkdude** [comment]
> Aww. Only after figuring this out I just realized there's https://github.com/babashka/neil/issues/162 with a comment. Just following and grabbing that code would've been maybe simpler 🤦

Do you still want give action to this or keep your current code?

**Comment by agzam** [comment]
> Do you still want give action to this or keep your current code?

I glanced over it, looks like it does pretty much almost the same thing. Either way, the bottom line - it works. If you have no objections, perhaps let's merge it and close both tickets. wdyt?

File: src/babashka/neil.clj [hunk 2]
```diff
@@ -500,22 +500,58 @@ will return libraries with 'test framework' in their description.
 See http://github.com/clojars/clojars-web/wiki/Search-Query-Syntax for
 details on the search syntax.")))
 
+(defn dep-search-maven [search-term]
+  (let [url (format
+             "https://search.maven.org/solrsearch/select?q=%s&rows=20&wt=json"
```
Comment (agzam) [inline src/babashka/neil.clj:505]: This number may seem speculative, no? I copypasted it as is from the documentation. Should we leave it at that or maybe increase it, I don't know to 50 or even more? But is it even sensible to try to get more results than this?

Comment (borkdude) [reply src/babashka/neil.clj:505]: Don't know to be honest

**Comment by borkdude** [comment]
I think adding a test here would be in order:

https://github.com/babashka/neil/blob/ff7aca40de042b398263d4cb6f9d77dbf73c5708/tests.clj#L78-L91

E.g. a test which searches for clojure on maven. After that, I think it'd be good to merge.

**Pull request merged**